### PR TITLE
Improve validation of children

### PR DESCRIPTION
**Pull request by westonruter**
This cherry-picks the commits by @miina from #1614 that relate specifically to the AMP spec and whitelist sanitizer. The intention here is that we can get this code merged into `develop` sooner for overall improvement to the plugin's AMP validation system, and leave the `amp-stories` branch purely for the UI for AMP stories.

### CODE REVIEW

**Comment by westonruter** [comment]
(Apparently the CLA bot doesn't like cherry-picked commits!)

File: tests/test-tag-and-attribute-sanitizer.php [hunk 1]
```diff
@@ -252,36 +258,55 @@ public function get_body_data() {
...
+							<amp-story-page id="my-second-page">
+								<i>bad</i>
+								<amp-analytics config="https://example.com/analytics.account.config.json"></amp-analytics>
+								<amp-story-grid-layer template="thirds">
+									<amp-img grid-area="bottom-third" src="https://example.ampproject.org/helloworld/bg2.gif" width="900" height="1600">
+									</amp-img>
+								</amp-story-grid-layer>
+								<amp-story-grid-layer template="vertical">
+									<h1>The End</h1>
+									<button>bad</button>
```
Comment (westonruter) [inline tests/test-tag-and-attribute-sanitizer.php]: This should be getting sanitized.

File: includes/sanitizers/class-amp-allowed-tags-generated.php [hunk 2]
```diff
@@ -2122,6 +2292,7 @@ class AMP_Allowed_Tags_Generated {
 							1,
 						),
 					),
+					'child_tags' => array(),
```
[the remark this answers is not in the document]
Comment (miina) [reply includes/sanitizers/class-amp-allowed-tags-generated.php]: Looks like the script is not accounting for `first_child_tag_name_oneof`:
https://github.com/ampproject/amphtml/blob/master/extensions/amp-geo/validator-amp-geo.protoascii#L35

File: composer.json [hunk 3]
```diff
@@ -5,6 +5,22 @@
...
+          "url": "https://github.com/ampproject/amphtml/archive/1811091519050.zip",
+          "type": "zip"
+        },
+        "source": {
+          "url": "git@github.com:ampproject/amphtml.git",
+          "type": "git",
+          "reference": "tags/1811091519050"
+        }
+      }
+    },
```
Comment (westonruter) [inline composer.json]: @felixarntz Any suggestion for a better way to do this?

File: includes/sanitizers/class-amp-allowed-tags-generated.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (westonruter) [reply includes/sanitizers/class-amp-allowed-tags-generated.php]: OK, this should be accounted for now.

**Comment by westonruter** [comment]
@miina How's this look now? If you can give it a ✅ approval that would be great.

**Pull request merged**